modules/openy_gc_shared_content/src/Plugin/SharedContentSourceType/VirtualYVideo.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace Drupal\openy_gc_shared_content\Plugin\SharedContentSourceType;
+
+use Drupal\openy_gc_shared_content\SharedContentSourceTypeBase;
+
+/**
+ * Defines the SharedContentSourceType plugin for Virtual Y video.
+ *
+ * @SharedContentSourceType(
+ *   id="gc_video",
+ *   label = @Translation("Virtual Y video"),
+ *   entityType="node",
+ *   entityBundle="gc_video"
+ * )
+ */
+class VirtualYVideo extends SharedContentSourceTypeBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getTeaserJsonApiQueryArgs() {
+    return [
+      'sort[sortByDate][path]' => 'created',
+      'sort[sortByDate][direction]' => 'DESC',
+      'filter[status]' => 1,
+      'filter[field_gc_share]' => 1,
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFullJsonApiQueryArgs() {
+    return [
+      'include' => 'field_gc_video_media,field_gc_video_media.thumbnail,field_gc_video_level,field_gc_video_category,field_gc_video_equipment',
```

Comment:
I think no need to move this to config because this is just a list of entity reference fields that should be included to request (for preview or save). This list will be changed only in case when we have new field for this content.